core/src/main/scala/chisel3/internal/BiConnect.scala

```diff
@@ -360,11 +401,11 @@ private[chisel3] object BiConnect {
       // Thus, right node better be a port node and thus have a direction hint
       ((left_direction, right_direction): @unchecked) match {
         //    CURRENT MOD   CHILD MOD
-        case (Input, Input)    => issueConnectL2R(left, right)
-        case (Internal, Input) => issueConnectL2R(left, right)
+        case (Input, Input)    => issueConnectL2R(left, lwr, right, rwr)
+        case (Internal, Input) => issueConnectL2R(left, lwr, right, rwr)
 
-        case (Output, Output)   => issueConnectR2L(left, right)
-        case (Internal, Output) => issueConnectR2L(left, right)
+        case (Output, Output)   => issueConnectR2L(left, lwr, right, rwr)
+        case (Internal, Output) => issueConnectR2L(left, lwr, right, rwr)
 
```

Comment:
The diff here can probably be made lower touch. The `issueConnect` functions call `.lref` so we could probably rely on the error reporting there, although that means it would construct wires in the error case (to provide a legal lref for a read-only thing) whereas doing it at this level allows us to elide issuing the connection altogether.

Separate from that, this code is also pretty inelegant in that it's checking `.topBinding` for each left and right multiple times. This may be optimized by the JIT, but it's pretty inefficient if it isn't.